src/lib/mina.ts

```
@@ -773,3 +805,143 @@ function defaultNetworkState(): NetworkValue {
+      case 'verificationKey':
+        return perm.setVerificationKey;
+      case 'permissions':
+        return perm.setPermissions;
+      case 'zkappUri':
+        return perm.setZkappUri;
+      case 'tokenSymbol':
+        return perm.setTokenSymbol;
+      case 'timing':
+        return 'None'; // TODO: do timing?
+      case 'votingFor':
+        return perm.setVotingFor; // TODO: do timing?
+      case 'sequenceEvents':
+        return perm.editSequenceState;
+      case 'incrementNonce':
+        return perm.incrementNonce;
+      case 'send':
+        return perm.send;
+      case 'receive':
+        return perm.receive;
+      default:
+        return '';
+    }
+  }
+
+  const update = accountUpdate.toJSON().body.update;
+
+  let isValidProof = false;
+  let isValidSignature = false;
+
+  // we don't check if proofs aren't enabled
+  if (!proofsEnabled) isValidProof = true;
+
+  if (accountUpdate.authorization.proof && proofsEnabled) {
+    let publicInput = accountUpdateToPublicInput(accountUpdate);
+    let publicInputFields = ZkappPublicInput.toFields(publicInput);
+
+    const proof = SmartContract.Proof().fromJSON({
+      maxProofsVerified: 2, // TODO: this can probably vary?
+      proof: accountUpdate.authorization.proof!,
+      publicInput: publicInputFields.map((f) => f.toString()),
+    });
+
+    try {
+      let verificationKey = account.zkapp?.verificationKey!;
+      isValidProof = await verify(proof, verificationKey);
+    } catch (error) {
+      isValidProof = false;
+    }
+  }
+
+  if (accountUpdate.authorization.signature) {
+    let txC = accountUpdate.body.useFullCommitment.toBoolean()
+      ? fullCommitment
+      : commitment;
+
+    // checking permissions and authorization for each party individually
+    try {
+      isValidSignature = Ledger.checkAccountUpdateSignature(
+        JSON.stringify(accountUpdate.toJSON()),
+        txC
+      );
+    } catch (error) {
+      isValidSignature = false;
```

same here about swallowing the error